src/rust/catnap/mod.rs

```diff
@@ -722,9 +722,14 @@ impl CatnapLibOS {
         self.runtime.free_sgarray(sga)
     }
 
-    /// Takes out the result from the [OperationTask] associated with the target [SchedulerHandle].
+    /// Takes out the result from the [OperationTask] associated with the target [TaskHandle].
     fn take_result(&mut self, handle: TaskHandle) -> (QDesc, OperationResult) {
-        let task: OperationTask = OperationTask::from(self.runtime.scheduler.remove(handle.clone()).as_any());
+        let task: OperationTask = if let Some(task) = self.runtime.scheduler.remove(&handle) {
+            OperationTask::from(task.as_any())
+        } else {
+            panic!("Removing task that does not exist (either was previously removed or never inserted)");
```

Comment:
Shouldn't this condition be `unreachable!()`? Even better, wouldn't be the case to change the function signature to return a `Fail` and we propagate it back to the PDPIX call?